Production_scripts/Create_costs/costs_district_nursing.R

```diff
@@ -0,0 +1,258 @@
+  mutate(year = convert_year_to_fyyear(contact_financial_year)) %>%
+  # rename TreatmentNHSBoardCode
+  rename(
+    hb2019 = treatment_nhs_board_code_9,
+    number_of_contacts = number_of_contacts
+  )
+
+# Join files together ------------------------------------------
+
+# match raw costs to contacts file
+dn_raw_costs_contacts <- full_join(dn_raw_contacts,
+  dn_raw_costs,
+  by = c("hb2019", "year")
+)
+
+
+# Deal with population cost-------------------------------------
+
+
+## Calculate population cost for NHS Highland with HSCP population ratio. ##
+# Of the two HSCPs, Argyll and Bute provides the
+# District Nursing data which is 27% of the population.
+
+population_lookup <- readr::read_rds(read_datazone_pop_file("HSCP2019_pop_est_1981_2020.rds")) %>%
+  # Select only the HSCPs for NHS Highland & years since 2015
+  filter(
+    hscp2019 %in% c("S37000004", "S37000016"),
+    year >= 2015
+  ) %>%
+  # Create year as FY = YYYY from CCYY.
+  rename(calendar_year = year) %>%
+  mutate(year = convert_year_to_fyyear(calendar_year)) %>%
+  group_by(year, hscp2019name) %>%
+  summarise(pop = sum(pop)) %>%
+  mutate(total_pop = sum(pop)) %>%
+  ungroup() %>%
+  # add Health Board code
+  mutate(hb2019 = "S08000022") %>%
+  ## compute proportion ##
+  mutate(
+    pop_proportion = pop / total_pop,
+    pop_pct = pop_proportion * 100
+  ) %>%
+  ## Argyll and Bute is the only HSCP in NHS Highland that submits data ##
+  filter(hscp2019name == "Argyll and Bute")
+
+
+# Join files -------------------------------------------
+
+## match files ##
+
+data <- full_join(dn_raw_costs_contacts,
+  population_lookup,
+  by = c("hb2019", "year")
+) %>%
+  # recode NA pop_proportion with 1
+  mutate(pop_proportion = replace_na(pop_proportion, 1)) %>%
+  ## total net cost ##
+  mutate(cost_total_net = ((cost * 1000) / (number_of_contacts / pop_proportion))) %>%
+  # sort by HB2019 and year
+  arrange(hb2019, year) %>%
+  # keep only records with cost
+  filter(!is.na(cost_total_net))
+
+
+# Fix incomplete submissions ------------------------------------------
+
+# If a Partnership has abnormally low contacts this will
+# affect the cost so use the previous year
+# until we have a complete submission
+
+## explore the trends
+
+data <-
+  data %>%
+  group_by(board_name) %>%
+  mutate(max_contacts = max(number_of_contacts)) %>%
+  mutate(pct_of_max = number_of_contacts / max_contacts * 100) %>%
+  ungroup()
+
+# plot #
+ggplot(data = data, aes(x = year, y = pct_of_max, group = board_name)) +
+  geom_line(aes(color = board_name)) +
+  labs(color = "NHS Board", x = "Year")
+
+
+# Deal with costs ------------------------------------------
+
+## costs with pct_of_max < 75 - uplift ##
+data <-
+  data %>%
+  group_by(board_name) %>%
+  arrange(year, .by_group = TRUE) %>%
+  mutate(cost_total_net = if_else(pct_of_max < 75,
+    lag(cost_total_net,
+      default = first(cost_total_net)
+    ) * 1.01,
+    cost_total_net
+  )) %>%
+  ungroup()
```

Comment:
Hi @catherineholland1 could you please look into this again? This is where the weird costs for highland are being produced. They are really high caused by the way the lag function is working here. If you look at the data the lag function is taking the row above if the percentage of contacts is lower than 75%. On one of the rows this gets 'fixed' but the row below is taking the costs before this is fixed, i.e the larger costs. Could you please see if you can find a way to fix this? Hope this makes sense. Thanks

Reply:
figured out an edit but requires some code to perhaps be run again if there are more than 1 suspect costs in the data - hopefully this fix makes sense and works!